.github/workflows/ci.yaml

```diff
@@ -1,5 +1,5 @@
 name: CI
-on: [push, pull_request, workflow_dispatch]
```

Comment:
Are you removing this because our tests are racey? Without each checking on each push, we might not catch errors that come up mid pr-review.

Reply:
We requires that tests are run after any code change on PR, and tests must be run on the latest changes before checkin.